graphql/src/execution/execution.rs

```diff
@@ -34,18 +40,22 @@ lazy_static! {
         .collect()
     };
 
-    // Cache expiry time, 1 minute by default.
-    static ref CACHE_EXPIRY: Duration = {
-        std::env::var("GRAPH_CACHE_EXPIRY_SECS")
-        .unwrap_or("60".to_string())
-        .parse::<u64>()
-        .map(|s| Duration::from_secs(s))
-        .unwrap()
+    static ref CACHE_ALL: bool = CACHED_SUBGRAPH_IDS.contains(&"*".to_string());
+
+    // How many blocks should be kept in the query cache. When the limit is reached, older blocks
+    // are evicted. This should be kept small since a lookup to the cache is O(n) on this value, and
+    // the cache memory usage also increases with larger number. Set to 0 to disable the cache,
+    // defaults to disabled.
+    static ref QUERY_CACHE_BLOCKS: usize = {
+        std::env::var("GRAPH_QUERY_CACHE_BLOCKS")
+        .unwrap_or("0".to_string())
```

Comment:
I would set the default to 1. (can be future PR). Setting CACHED_SUBGRAPH_IDS and not getting a cache is surprising behavior.

Reply:
Yes, this is envisioning a future where the default of `CACHED_SUBGRAPH_IDS` is `*`. So that this is an on/off switch for operators, manually adding ids is not fun.